pandas/plotting/_matplotlib/core.py

```diff
@@ -234,6 +249,52 @@ def _validate_color_args(self):
                             "pass 'style' without a color symbol"
                         )
 
+    @staticmethod
+    def _create_iter_data_given_by(
+        data: ABCDataFrame, by: Optional[list]
```

Comment:
```suggestion
        data: DataFrame, by: Optional[list]
```

The ABC classes don't work well as annotations since they aren't declared statically, so should opt for the actual objects here and on subsequent line

Reply:
thanks! changed for all used cases in here